client/asset/dcr/spv.go

```diff
@@ -0,0 +1,1020 @@
+		Spendable:               bal.Spendable.ToCoin(),
+		Total:                   bal.Total.ToCoin(),
+		Unconfirmed:             bal.Unconfirmed.ToCoin(),
+		VotingAuthority:         bal.VotingAuthority.ToCoin(),
+	}, nil
+}
+
+// LockedOutputs fetches locked outputs for the specified account.
+// Part of the Wallet interface.
+func (w *spvWallet) LockedOutputs(ctx context.Context, _ string) ([]chainjson.TransactionInput, error) {
+	return w.dcrWallet.LockedOutpoints(ctx, w.acctName)
+}
+
+// Unspents fetches unspent outputs for the specified account.
+// Part of the Wallet interface.
+func (w *spvWallet) Unspents(ctx context.Context, _ string) ([]*walletjson.ListUnspentResult, error) {
+	return w.dcrWallet.ListUnspent(ctx, 0, math.MaxInt32, nil, w.acctName)
+}
+
+// LockUnspent locks or unlocks the specified outpoint.
+// Part of the Wallet interface.
+func (w *spvWallet) LockUnspent(ctx context.Context, unlock bool, ops []*wire.OutPoint) error {
+	if unlock {
+		for _, op := range ops {
+			w.UnlockOutpoint(&op.Hash, op.Index)
+		}
+		return nil
+	}
+	for _, op := range ops {
+		w.LockOutpoint(&op.Hash, op.Index)
+	}
+
+	return nil
+}
+
+// UnspentOutput returns information about an unspent tx output, if found
+// and unspent.
+// This method is only guaranteed to return results for outputs that pay to
+// the wallet. Returns asset.CoinNotFoundError if the unspent output cannot
+// be located.
+// Part of the Wallet interface.
+func (w *spvWallet) UnspentOutput(ctx context.Context, txHash *chainhash.Hash, index uint32, _ int8) (*TxOutput, error) {
+	txd, err := w.dcrWallet.TxDetails(ctx, txHash)
+	if errors.Is(err, walleterrors.NotExist) {
+		return nil, asset.CoinNotFoundError
+	} else if err != nil {
+		return nil, err
+	}
+
+	details, err := w.ListTransactionDetails(ctx, txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	var addrStr string
+	for _, detail := range details {
+		if detail.Vout == index {
+			addrStr = detail.Address
+		}
+	}
+	if addrStr == "" {
+		return nil, fmt.Errorf("error locating address for output")
+	}
+
+	tree := wire.TxTreeRegular
+	if txd.TxType != stake.TxTypeRegular {
+		tree = wire.TxTreeStake
+	}
+
+	if len(txd.MsgTx.TxOut) <= int(index) {
+		return nil, fmt.Errorf("not enough outputs")
+	}
+
+	_, tipHeight := w.MainChainTip(ctx)
+
+	var ours bool
+	for _, credit := range txd.Credits {
+		if credit.Index == index {
+			if credit.Spent {
+				return nil, asset.CoinNotFoundError
+			}
+			ours = true
+			break
+		}
+	}
+
+	if !ours {
+		return nil, asset.CoinNotFoundError
+	}
+
+	return &TxOutput{
+		TxOut:         txd.MsgTx.TxOut[index],
+		Tree:          tree,
+		Addresses:     []string{addrStr},
+		Confirmations: uint32(txd.Block.Height - tipHeight + 1),
+	}, nil
+}
+
+// ExternalAddress returns an external address using GapPolicyIgnore.
+// Part of the Wallet interface.
+//
+// NOTE: Why GapPolicyIgnore instead of GapPolicyWrap? Can we recover from an
+// ignored gap policy without extreme measures?
```

Comment:
Maybe we are trying a little to hard to avoid address re-use.  In the Electrum PR, I spent some time looking at where we are burning through addresses, and I decided to [add a `refundAddress` method to `client/asset/btc.Wallet`](https://github.com/decred/dcrdex/pull/1607/commits/c63ef108d423cd5ffe64de25a8d8b5fd68f38915#diff-1d84027a7ea77f3c9b4b3adea610bde93e9a489d8e8f1f977b8121699c8af0fdR32-R34) where the addresses are very rarely used and we can tolerate some reuse.

The other big source of wasted addresses is the address provided in orders, when the address really [could be provided closer to match time](https://github.com/decred/dcrdex/pull/1607/commits/c63ef108d423cd5ffe64de25a8d8b5fd68f38915#diff-f750305f8da9a4959a035c59e067874c849530184462549bbe658a8a79a89e5fR424) although with more messaging, so that is not as easy.

Regarding recovery, when we implement Rescanner, we can use the DiscoverAddresses method with a provided gap limit.  Also, I haven't gotten far enough with the review, but we should ensure the seed can be used in an external dcrwallet for manual recovery.